### PR TITLE
ref(audit-log): log raw dynamic sampling changes - [TET-341]

### PR DESCRIPTION
**Pull request by priscilawebdev**
Before we were logging just 'edited project settings' and now we log the whole updated dynamic sampling object

### CODE REVIEW

File: src/sentry/api/endpoints/project_details.py [hunk 1]
```diff
@@ -647,6 +647,7 @@ def put(self, request: Request, project) -> Response:
             raw_dynamic_sampling = result["dynamicSampling"]
             fixed_rules = self._fix_rule_ids(project, raw_dynamic_sampling)
             project.update_option("sentry:dynamic_sampling", fixed_rules)
```
Comment (ahmedetefy) [inline src/sentry/api/endpoints/project_details.py:649]:
```suggestion
            if project.update_option("sentry:dynamic_sampling", fixed_rules):
                changed_proj_settings["sentry:dynamic_sampling"] = result["dynamicSampling"]
```

File: src/sentry/api/endpoints/project_details.py [hunk 2]
```diff
@@ -646,7 +646,8 @@ def put(self, request: Request, project) -> Response:
         if "dynamicSampling" in result:
             raw_dynamic_sampling = result["dynamicSampling"]
             fixed_rules = self._fix_rule_ids(project, raw_dynamic_sampling)
-            project.update_option("sentry:dynamic_sampling", fixed_rules)
+            if project.update_option("sentry:dynamic_sampling", fixed_rules):
+                changed_proj_settings["sentry:dynamic_sampling"] = result["dynamicSampling"]
             changed_proj_settings["sentry:dynamic_sampling"] = result["dynamicSampling"]
```
Comment (ahmedetefy) [inline src/sentry/api/endpoints/project_details.py:651]: Remove this line please as it is repeated

Comment (priscilawebdev) [reply src/sentry/api/endpoints/project_details.py]: just did 😉

**Review by ahmedetefy — APPROVED** [review]
Thanks

**Review by andriisoldatenko — APPROVED** [review]

**Pull request merged**